src/lib/components/Tooltip.svelte

```diff
@@ -1,80 +1,68 @@
 <script lang="ts">
   import { onDestroy } from "svelte";
   import { debounce } from "@dfinity/utils";
+  import { translateTooltip } from "$lib/utils/tooltip.utils";
 
   export let id: string;
   export let testId = "tooltip-component";
   export let text = "";
   export let noWrap = false;
   export let top = false;
   export let center = false;
-  export let containerSelector = "main";
 
   let tooltipComponent: HTMLDivElement | undefined = undefined;
   let target: HTMLDivElement | undefined = undefined;
   let innerWidth: number | undefined = undefined;
+  let tooltipTransformX: number = 0;
+  let tooltipTransformY: number = 0;
   let tooltipStyle: string | undefined = undefined;
 
-  const setPosition = debounce(() => {
+  $: tooltipStyle = `--tooltip-transform-x: ${tooltipTransformX}px; --tooltip-transform-y: ${tooltipTransformY}px;`;
+
+  // Finds the first ancestor with hidden horizontal overflow.
+  const getOverflowContainer = (element: HTMLElement): HTMLElement => {
```

Comment:
this function can potentially return null no? if an HTMLElement is added to the dom but not yet rendered its parent is undefined (or null don't remember)